File: pset6/readability/readability.py
from cs50 import get_string

# Get text from user
text = get_string("Text: ")

letters = 0
words = 0
sentences = 0

# Count number of words, letters, sentences in text
for i, j in enumerate(text):
    if (j.isalpha()):
        letters += 1

    if j.isspace() or i == len(text) - 1:
        words += 1

    if j in ['.', '!', '?']:
        sentences += 1

L = letters / words * 100
S = sentences / words * 100

# Coleman-Liau index
grade = round(0.0588 * L - 0.296 * S - 15.8)

# Print grade
if grade > 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade " + str(grade))

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    // Count number of letters
    int letters = 0;
    int words = 0;
    int sentences = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }

        if (isspace(text[i]) || i == strlen(text) - 1)
        {
            words++;
        }

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    float L, S;

    L = (float) letters / words * 100;
    S = (float) sentences / words * 100;

    int grade;
    grade = round(0.0588 * L - 0.296 * S - 15.8);

    if (grade > 16)
    {
        printf("Grade 16+\n");
    }

    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }

    return 0;
}